cmake/CMakeLists.txt

```
@@ -241,6 +241,9 @@ option(onnxruntime_ENABLE_CPUINFO "Enable cpuinfo" ON)
 # ATen fallback support
 option(onnxruntime_ENABLE_ATEN "Enable ATen fallback" OFF)
 
+# dlpack support
+option(onnxruntime_ENABLE_DLPACK "Enable dlpack" OFF)
```

Removing the option in build.py and making this to dependent option seems to be more optimal to satisfy the CI.
```suggestion
cmake_dependent_option(onnxruntime_ENABLE_DLPACK "Enable dlpack" ON "onnxruntime_ENABLE_TRAINING 
 OR onnxruntime_ENABLE_ATEN OR onnxruntime_ENABLE_PYTHON" OFF)
```